h2o-core/src/test/java/water/util/FileUtilsTest.java

```diff
@@ -0,0 +1,68 @@
+package water.util;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.runner.RunWith;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+@RunWith(H2ORunner.class)
+@CloudSize(1)
+public class FileUtilsTest {
+
+    @Rule
+    public EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+    @Test
+    public void testFindFileInPredefinedPath() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        final Method findFileInPredefinedPath = FileUtils.class.getDeclaredMethod("findFileInPredefinedPath",
+                String.class);
+        assertTrue(Modifier.isStatic(findFileInPredefinedPath.getModifiers()));
+        assertTrue(Modifier.isPrivate(findFileInPredefinedPath.getModifiers()));
+        try {
+            final File h2oHomeDir = new File(System.getProperty("user.dir")).getParentFile();
+            environmentVariables.set("H2O_FILES_SEARCH_PATH", h2oHomeDir.getAbsolutePath());
+
+            findFileInPredefinedPath.setAccessible(true);
```

Comment:
you could leave the method as package private and have easier time testing it

Reply:
I could, but this is easy enough. The method should stay private. Bending code because of JUnit test is only replacing one smell with another. The code that's actually used has proper visibility and the test is adjusted. I'd much rather write a slightly more complex test than exposing things with wrong visibility in production code.